src/AWS/Lambda/KinesisDataStreamsEvent.hs

```diff
@@ -1,69 +1,56 @@
-
-module AWS.Lambda.KinesisDataStreamsEvent
-  (
-    KinesisDataStreamsEvent(..),
-    Record(..),
-    Kinesis(..)
-  ) where
+{-# language DeriveGeneric   #-}
+{-# language RecordWildCards #-}
+{-# language TemplateHaskell #-}
+module AWS.Lambda.KinesisDataStreamsEvent where
 
+import Control.Lens
 import Data.Aeson
 import Data.HashMap.Strict
 import Data.Text (Text)
 import GHC.Generics
 
-data Kinesis =
-  Kinesis {
-    partitionKey         :: Text,
-    payload              :: Text,
-    kinesisSchemaVersion :: Text,
-    sequenceNumber       :: Text
-  } deriving (Show)
+data Kinesis = Kinesis
+  { _partitionKey         :: Text
+  , _payload              :: Text
+  , _kinesisSchemaVersion :: Text
+  , _sequenceNumber       :: Text
+  } deriving (Eq, Generic, Show)
+
+instance ToJSON Kinesis where
+  toJSON = genericToJSON defaultOptions { fieldLabelModifier = drop 1 }
 
 instance FromJSON Kinesis where
-  parseJSON (Object v) = Kinesis
-    <$> v .: "partitionKey"
-    <*> v .: "data"
-    <*> v .: "kinesisSchemaVersion"
-    <*> v .: "sequenceNumber"
+  parseJSON = genericParseJSON defaultOptions { fieldLabelModifier = drop 1 }
 
-instance ToJSON Kinesis where
-  toJSON Kinesis{..} =
-    object [
-        "partitionKey" .= partitionKey
-      , "data" .= payload
-      , "kinesisSchemaVersion" .= kinesisSchemaVersion
-      , "sequenceNumber" .= sequenceNumber
-    ]
 
-data Record =
-  Record {
-    eventID           :: Text,
-    eventVersion      :: Text,
-    kinesis           :: Kinesis,
-    invokeIdentityArn :: Text,
-    eventName         :: Text,
-    eventSourceARN    :: Text,
-    eventSource       :: Text,
-    awsRegion         :: Text
-  } deriving (Show, Generic)
+data Record = Record
+  { _eventID           :: Text
+  , _eventVersion      :: Text
+  , _kinesis           :: Kinesis
+  , _invokeIdentityArn :: Text
+  , _eventName         :: Text
+  , _eventSourceARN    :: Text
+  , _eventSource       :: Text
+  , _awsRegion         :: Text
+  } deriving (Eq, Generic, Show)
 
-instance FromJSON Record
-instance ToJSON Record
+instance ToJSON Record where
+  toJSON = genericToJSON defaultOptions { fieldLabelModifier = drop 1 }
 
-data KinesisDataStreamsEvent =
-  KinesisDataStreamsEvent {
-    records :: [Record]
-  }
+instance FromJSON Record where
+  parseJSON = genericParseJSON defaultOptions { fieldLabelModifier = drop 1 }
 
-instance FromJSON KinesisDataStreamsEvent where
-  parseJSON = withObject "KinesisDataStreamsEvent" $
-    \v -> KinesisDataStreamsEvent
-      <$> ((v .: "Records") >>= parseJSONList)
 
 instance ToJSON KinesisDataStreamsEvent where
-  toJSON (KinesisDataStreamsEvent{..}) =
-    object ["Records" .= records]
+  toJSON = genericToJSON defaultOptions { fieldLabelModifier = drop 1 }
+
+instance FromJSON KinesisDataStreamsEvent where
+  parseJSON = genericParseJSON defaultOptions { fieldLabelModifier = drop 1 }
+
+newtype KinesisDataStreamsEvent = KinesisDataStreamsEvent { _records :: [Record] }
```

Comment:
this needs to be uppercase in the JSON

Reply:
Sorry, I'm not sure what "this" is referring to.

Reply:
`records`